scipy/stats/_continuous_distns.py

```diff
@@ -1489,8 +1489,20 @@ def _stats(self, df):
 
     def _entropy(self, df):
         half_df = 0.5 * df
-        return (half_df + np.log(2) + sc.gammaln(half_df) +
-                (1 - half_df) * sc.psi(half_df))
+        return _lazywhere(half_df < 1e12, (half_df, ),
+                          lambda half_df: (half_df + np.log(2) +
+                                           sc.gammaln(half_df) +
+                                           (1 - half_df) *
+                                           sc.psi(half_df)),
+                          # plug in the above formula the following asymptotic
+                          # expansions:
+                          # ln(gamma(a)) ~ (a - 0.5) * ln(a) - a +
+                          #                 0.5 * ln(2 * pi)
+                          # psi(a) ~ ln(a) - 1/(2 * a)
+                          f2=lambda half_df: (np.log(2) + 0.5 *
+                                              (1. + np.log(half_df) +
+                                               np.log(2*np.pi)))
+                          )
```

Comment:
Can you try this out when including the $O\left(\frac{1}{a}\right)$ term that you can get by using the longer expansions from https://github.com/scipy/scipy/pull/17929#issuecomment-1419505605?

Reply:
Using the longer expansion yields a formula which is very accurate already for small values of $df$:

![Chi2_Entropy_Small_Expansion](https://user-images.githubusercontent.com/40656107/217642922-3e771beb-7081-47cc-8a98-97338100113b.png)

<details>

import numpy as np
import matplotlib.pyplot as plt
from mpmath import mp
from scipy import stats
from scipy import special as sc

def entropy_chi2(df):
    half_df = 0.5 * df
    return np.where(half_df < 0,
                    half_df + np.log(2) + sc.gammaln(half_df) +
                    (1 - half_df) * sc.psi(half_df),
                    np.log(2) + 0.5 * (1. + np.log(half_df) + np.log(2*np.pi)) - 2/(3 * df) - 1/(3 * df**2))

def entropy_chi2_naive(df):
    half_df = 0.5 * df
    return half_df + np.log(2) + sc.gammaln(half_df) + (1 - half_df) * sc.psi(half_df)

@np.vectorize
def entropy_mpmath(df):
    mp.dps = 100
    df = mp.mpf(df)
    half_df = df / 2
    return half_df + mp.log(2) + mp.log(mp.gamma(half_df)) + (1 - half_df) * mp.digamma(half_df)

x = np.linspace(1e-8, 3, 500)
plt.plot(x, entropy_chi2(x), label="Expansion", ls="dashed")
plt.plot(x, entropy_chi2_naive(x), label="main", ls="dotted")
plt.plot(x, np.asarray(entropy_mpmath(x), np.float64), label="mpmath", ls="dashdot")
plt.ylim(-3, 3)
plt.legend()
plt.title("$\chi^2$ entropy")

</details>